packages/server/src/sdk/app/rows/search.ts

```diff
@@ -73,6 +76,24 @@ export async function search(
     const table = await sdk.tables.getTable(options.tableId)
     options = searchInputMapping(table, options)
 
+    const visibleTableFields = Object.keys(table.schema).filter(
+      f => table.schema[f].visible !== false
+    )
+
+    if (options.fields) {
```

Comment:
I think this may be too high level for this - I *think* the SQL layers may ignore/override this in terms of query generation. If you look at `server/src/api/controllers/row/ExternalRequest.ts` on line 692 there is a function called `buildSqlFieldList` -  this comes up with a list of fields which are going to be defined as part of the `SELECT <fields>` in the query. SQS does something similar with its function `buildInternalFieldList` - these today are independent of each other as how they generate their list of fields is quite different.

We could merge these into one path through the SDK which fetches a list of fields (and relationships) to add to the `QueryJson` which would take into account if it was performed through a view - we'd need to pass the list of `visibleTableFields` down to the search implementations via the `RowSearchParams`.

Reply:
We would also need to handle the list of relationships, found top level on the `QueryJson` structure under the `relationships` key - this would need to have relationships removed from it if the relationship column is disabled through the view (this removes the `JOIN`). There are again two functions which do this today, for external DBs `buildExternalRelationships` and for SQS `buildInternalRelationships`.

Reply:
NOTE: none of this can be applicable to Lucene as it always returns the full row JSON, it cannot strip fields out at the database level - I wouldn't worry about this too much.

Reply:
After a reanalising the logic, I moved the trim to the outputProcessing. We still need to fetch all the columns for things like formulas.
This PR was meant to cover only the security around queries, so I will stick to it and I will tackle the actual SQL statement generation in a separated PR as talked on the meeting this morning